shared/src/commonMain/kotlin/io.newm.shared/internal/implementations/WalletNFTSongsUseCaseImpl.kt

```diff
@@ -2,19 +2,23 @@ package io.newm.shared.internal.implementations
 
 import io.newm.shared.internal.repositories.CardanoWalletRepository
 import io.newm.shared.internal.repositories.ConnectWalletManager
-import io.newm.shared.internal.repositories.testdata.MockSongs
 import io.newm.shared.public.models.NFTTrack
 import io.newm.shared.public.models.error.KMMException
-import io.newm.shared.public.usecases.WalletNFTSongsUseCase
+import io.newm.shared.public.usecases.WalletNFTTracksUseCase
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
 
 internal class WalletNFTSongsUseCaseImpl(
     private val cardanoRepository: CardanoWalletRepository,
     private val connectWalletManager: ConnectWalletManager
-) : WalletNFTSongsUseCase {
+) : WalletNFTTracksUseCase {
 
-    override fun getAllWalletNFTSongs(): Flow<List<NFTTrack>> = MockSongs.getSongsFlow()
-    override suspend fun getWalletNFTs(): List<NFTTrack> {
+    override fun getAllNFTTracksFlow(): Flow<List<NFTTrack>> {
+        return flow {
```

Comment:
Can these flows be moved to the callsite in the view models since we can't use them on iOS, or does it need to be here for some reason?

Reply:
We need them here, there will a web client down the road that can be power through KMM. Ideally once we figure out how to consume flows on iOS we can move to a more reactive flow.